src/tokenserver/db/models.rs

```diff
@@ -61,152 +70,294 @@ impl TokenserverDb {
         }
     }
 
-    fn get_user_sync(&self, email: String) -> DbResult<results::GetUser> {
-        let query = r#"
-            SELECT users.uid, users.email, users.client_state, users.generation,
-                   users.keys_changed_at, users.created_at, nodes.node
-              FROM users
-              JOIN nodes
-                ON nodes.id = users.nodeid
-             WHERE users.email = ?
+    /// Get the most current user record for the given email and service ID. This function also
+    /// marks any old user records as replaced, in case of data races that may have occurred
+    /// during row creation.
+    fn get_user_sync(&self, params: params::GetUser) -> DbResult<results::GetUser> {
+        const QUERY: &str = r#"
```

Comment:
I updated this query to more closely match the query used on the legacy Tokenserver. The analogous database method on the legacy Tokenserver is [here](https://github.com/mozilla-services/tokenserver/blob/master/tokenserver/assignment/sqlnode/sql.py#L285)

Reply:
well, [here](https://github.com/mozilla-services/tokenserver/blob/master/tokenserver/assignment/sqlnode/sql.py#L44)

Reply:
I was linking to the actual Python methods that served as inspiration for these Rust methods, but yeah, I should probably have included links to the queries as well